File: projetDoki/src/main/java/formation/sopra/bu/projetDoki/metier/Praticien.java
package formation.sopra.bu.projetDoki.metier;

import java.util.List;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonView;

import formation.sopra.bu.projetDoki.view.JsonViews;

@Entity
@Table(name = "praticien")
@DiscriminatorValue("praticien")
public class Praticien extends Personne {

//Attributs
	@OneToMany(mappedBy = "key.praticien")
	@Column(name = "specialite")
	@JsonView(JsonViews.PraticienAvecSpe.class)
	private List<PraticienSpecialite> specialites;

	@OneToMany(mappedBy = "key.praticien")
	private List<PraticienMotif> motifs;

	@Column(name = "disponibilites")
	@OneToMany
	private List<Disponibilite> dispos;

	@OneToMany(mappedBy = "praticien") // , fetch = FetchType.LAZY) ???
	@Column(name = "rdvs_avec-patient")
	@JsonView(JsonViews.PraticienAvecRdv.class)
	private List<RendezVous> rdvs;

//Constructeurs
	public Praticien() {
	}

	public Praticien(String mail, String username, String password, String civilite, String nom, String prenom,
			String telephone, List<PraticienSpecialite> specialites, List<PraticienMotif> motifs,
			List<Disponibilite> dispos, List<RendezVous> rdvs, boolean enable, List<UserRole> roles, int version) {
		super(mail, username, password, civilite, nom, prenom, telephone, enable, roles, version);
		this.specialites = specialites;
		this.motifs = motifs;
		this.dispos = dispos;
		this.rdvs = rdvs;
	}

//Getters & Setters

	public List<PraticienMotif> getMotifs() {
		return motifs;
	}

	public List<PraticienSpecialite> getSpecialites() {
		return specialites;
	}

	public void setSpecialites(List<PraticienSpecialite> specialites) {
		this.specialites = specialites;
	}

	public void setMotifs(List<PraticienMotif> motifs) {
		this.motifs = motifs;
	}

	public List<Disponibilite> getDispos() {
		return dispos;
	}

	public void setDispos(List<Disponibilite> dispos) {
		this.dispos = dispos;
	}

	public List<RendezVous> getRdvs() {
		return rdvs;
	}

	public void setRdvs(List<RendezVous> rdvs) {
		this.rdvs = rdvs;
	}

//HashCode & Equals
	public int hashCode() {
		final int prime = 31;
		int result = super.hashCode();
		result = prime * result + ((dispos == null) ? 0 : dispos.hashCode());
		result = prime * result + ((motifs == null) ? 0 : motifs.hashCode());
		result = prime * result + ((rdvs == null) ? 0 : rdvs.hashCode());
		result = prime * result + ((specialites == null) ? 0 : specialites.hashCode());
		return result;
	}

	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (!super.equals(obj))
			return false;
		if (getClass() != obj.getClass())
			return false;
		Praticien other = (Praticien) obj;
		if (dispos == null) {
			if (other.dispos != null)
				return false;
		} else if (!dispos.equals(other.dispos))
			return false;
		if (motifs == null) {
			if (other.motifs != null)
				return false;
		} else if (!motifs.equals(other.motifs))
			return false;
		if (rdvs == null) {
			if (other.rdvs != null)
				return false;
		} else if (!rdvs.equals(other.rdvs))
			return false;
		if (specialites == null) {
			if (other.specialites != null)
				return false;
		} else if (!specialites.equals(other.specialites))
			return false;
		return true;
	}

}

File: Doki/angular/src/app/model/praticien.ts
import {RendezVous} from './rendez-vous';
import {Motif} from './motif';
import {Specialite} from './specialite';
import {Disponibilite} from './disponibilite';
import {Personne} from './personne';
import {Adresse} from './adresse';

export class Praticien extends Personne {

  private _cabinet: Adresse;
  private _specialites: Array<Specialite> = new Array<Specialite>();
  private _dispos: Array<Disponibilite> = new Array<Disponibilite>();
  private _rdvs: Array<RendezVous> = new Array<RendezVous>();
  private _motifs: Array<Motif> = new Array<Motif>();

  constructor(mail?: string, username?: string, password?: string, civilite?: string, nom?: string, prenom?: string, telephone?: string,
              cabinet?: Adresse, specialites?: Array<Specialite>, motifs?: Array<Motif>, dispos?: Array<Disponibilite>,
              rdvs?: Array<RendezVous>) {
    super(mail, username, password, civilite, nom, prenom, telephone);
    this._cabinet = cabinet;
    this._specialites = specialites;
    this._dispos = dispos;
    this._rdvs = rdvs;
    this._motifs = motifs;
  }

  get cabinet(): Adresse {
    return this._cabinet;
  }

  set cabinet(value: Adresse) {
    this._cabinet = value;
  }

  get specialites(): Array<Specialite> {
    return this._specialites;
  }

  set specialites(value: Array<Specialite>) {
    this._specialites = value;
  }

  get dispos(): Array<Disponibilite> {
    return this._dispos;
  }

  set dispos(value: Array<Disponibilite>) {
    this._dispos = value;
  }

  get rdvs(): Array<RendezVous> {
    return this._rdvs;
  }

  set rdvs(value: Array<RendezVous>) {
    this._rdvs = value;
  }

  get motifs(): Array<Motif> {
    return this._motifs;
  }

  set motifs(value: Array<Motif>) {
    this._motifs = value;
  }
}
